### PR TITLE
Erin/studio assessment controller

### PR DESCRIPTION
**Pull request by erinysong**
## Studio Assessment Controller

Created Studio Assessment Controller that performs CRUD functionality (similar to casenotes and paperwork)
Also changed some fields to the Studio Assessment Table but if there are any issues with that feel free to let me know

### Related PRs

Optional - any related PRs you're waiting on, or PRs that will conflict, etc

### Migrations

Optional - if you added anything to the database through migration(s)

### Screenshots

Optional but would be nice for final presentation :)

CC: @didvi

### CODE REVIEW

File: app/controllers/api/studio_assessments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        end
+    end
+
+    def user_not_authorized
+        render json: { error: 'You are not authorized to perform this action' }, status: :unauthorized
+    end
+
+    private
+
+    def set_studio_assessment
```
Comment (julianrkung) [inline app/controllers/api/studio_assessments_controller.rb]: Divi and I talked about not doing authorization in set_studio_assessment and instead moving the authorization to each of the individual CRUD functions (so adding an authorize call to show, update, and destroy). I think that makes more sense? Would like your input! If we decide to do authorization in set_studio_assessments I can update the studio_asessment_policy pundit though. :)

File: db/migrate/20200307003647_create_studio_assessments.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      t.string :react_comment
+      t.integer :node_score
+      t.string :node_comment
+      t.integer :db_score
+      t.string :db_comment
+      t.integer :problemsolving_score
+      t.string :problemsolving_comment
+      t.integer :problemsolvingalt_score
+      t.string :problemsolvingalt_comment
+      t.boolean :passed_capstone
```
Comment (julianrkung) [inline db/migrate/20200307003647_create_studio_assessments.rb]: maybe `capstone_passed` to keep consistent w/above styling. If we change this we need to change it in the api controller and views and others also!

File: db/schema.rb [hunk 3]
```diff
@@ -93,6 +93,34 @@
...
+    t.string "react_comment"
+    t.integer "node_score"
+    t.string "node_comment"
+    t.integer "db_score"
+    t.string "db_comment"
+    t.integer "problemsolving_score"
+    t.string "problemsolving_comment"
+    t.integer "problemsolvingalt_score"
+    t.string "problemsolvingalt_comment"
+    t.boolean "passed_capstone"
```
Comment (julianrkung) [inline db/schema.rb]: `capstone_passed`

File: app/assets/stylesheets/scaffolds.scss [hunk 4]
```diff
@@ -0,0 +1,84 @@
+body {
```
[the remark this answers is not in the document]
Comment (erinysong) [reply app/assets/stylesheets/scaffolds.scss]: think divi ended up deleting this file and it seems to be gone after i pulled from master so thanks for pointing this out! def should have deleted all the unnecessary files that scaffold generated

File: app/controllers/api/studio_assessments_controller.rb [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        @studio_assessment = authorize StudioAssessment.find(params[:id])
+    rescue ActiveRecord::RecordNotFound
+        render json: { error: 'Could not find studio assessment' }, status: :not_found
+    end
+
+    def studio_assessment_params
+        case_notes_param = params.require(:studio_assessment).permit(:participant_id, 
+                                                                    :staff_id,
```
[the remark this answers is not in the document]
Comment (erinysong) [reply app/controllers/api/studio_assessments_controller.rb]: yeah you're right I completely overlooked deleting that field after I added the merge line - just fixed it, thanks for catching that :))

**Comment by erinysong** [comment]
addressed all the comments made, feel free to let me know if there was anything I missed but I made the following changes:

removed unnecessary extra staff_id field since we have a line that merges it from the staff table anyway
moved authorization to the individuals CRUD functions in studio assessments controller
changed passed_capstone field in studio assessments db to capstone_passed for consistence

**Comment by erinysong** [comment]
just made the changes that you requested above, thanks divi! :)

**Pull request merged**